From 82e1cfe41dc0c209861c805503e9cb41111eb2ed Mon Sep 17 00:00:00 2001
From: Tony <twickham@google.com>
Date: Sun, 22 Jul 2018 13:07:04 -0700
Subject: [PATCH 6/9] Fix gap between notification dots and their shadows

There was a rounding issue due to usnig integer division when we meant float

Bug: 73331123
Change-Id: Ic3deb1c26aa00808a929e1303602656ce532963d
---
 src/com/android/launcher3/graphics/ShadowGenerator.java | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/com/android/launcher3/graphics/ShadowGenerator.java b/src/com/android/launcher3/graphics/ShadowGenerator.java
index 5fbf502aa..88da853d6 100644
--- a/src/com/android/launcher3/graphics/ShadowGenerator.java
+++ b/src/com/android/launcher3/graphics/ShadowGenerator.java
@@ -126,13 +126,13 @@ public class ShadowGenerator {
         }
 
         public Bitmap createPill(int width, int height) {
-            radius = height / 2;
+            radius = height / 2f;
 
-            int centerX = Math.round(width / 2 + shadowBlur);
+            int centerX = Math.round(width / 2f + shadowBlur);
             int centerY = Math.round(radius + shadowBlur + keyShadowDistance);
             int center = Math.max(centerX, centerY);
             bounds.set(0, 0, width, height);
-            bounds.offsetTo(center - width / 2, center - height / 2);
+            bounds.offsetTo(center - width / 2f, center - height / 2f);
 
             int size = center * 2;
             Bitmap result = Bitmap.createBitmap(size, size, Config.ARGB_8888);
-- 
2.17.1

